aten/src/ATen/cuda/CUDAGraph.cu

```diff
@@ -0,0 +1,38 @@
+#include <ATen/cuda/CUDAGraph.h>
+
+namespace at::cuda {
+
+__global__ void dynamic_graph_updater_kernel(cudaGraphKernelNodeUpdate* updates, size_t num_updates) {
+    constexpr size_t buf_sz = 1024;
+    __shared__ const void* pointer_indirection[buf_sz];
+    for (size_t off = 0; off < num_updates; off += buf_sz) {
+        int local_num_updates = min(buf_sz, num_updates - off);
+        for (int i = 0; i < local_num_updates; i++) {
+            cudaGraphKernelNodeUpdate& this_update = updates[i + off];
+            if (this_update.field == cudaGraphKernelNodeFieldParam) {
+                if (this_update.updateData.param.size != sizeof(void*)) {
+                    printf("dynamic graph updater only supports pointers\n");
+                    __trap();
+                }
+                // assume pValue points directly to the data
+                pointer_indirection[i] = this_update.updateData.param.pValue;
+                // pointer_indirection[i] now points to the data
+                // &pointer_indirection[i] now points to a void* that points to the data
+                this_update.updateData.param.pValue = &pointer_indirection[i];
+                // now the update will read *&pointer_indirection[i], which gives the original pValue, and splat that into the correct kernel arg
+            }
+        }
+        cudaError_t error = cudaGraphKernelNodeUpdatesApply(&updates[off], local_num_updates);
```

Comment:
Do you think it could be worth sorting / grouping the `cudaGraphKernelNodeUpdate`s? Perhaps putting together all of those pertaining to the same kernel launch?

Reply:
yes, that is probably beneficial. But TBH, I would say don't worry about this right now.